### PR TITLE
Tidy up protobuf

### PR DESCRIPTION
**Pull request by benson31**
There's an old antipattern in our protobuf messages that uses ASCII strings to represented vector data. This was convenient in the days of hand-written prototext, but now that the PFE writes our protobuf messages, it's less optimal and often confusing (e.g, "Why is the `dims` parameter the string `'1 1 1'` and not a list of dimensions?"). Additionally, this isolates and formalizes some of our protobuf utilities. The large file count is largely due to cleaning `str_list` out of the unit tests.

[CI Pipeline](https://lc.llnl.gov/gitlab/benson31/tomslbannfork/-/pipelines/165668)

### CODE REVIEW

File: applications/ATOM/models/wae.py [hunk 1]
```diff
@@ -254,15 +250,15 @@ def forward_decoder(self, x_emb, z):
...
+            dims=[self.input_feature_dims, self.zdim],
         )
         x_input = lbann.Concatenation(x_emb, z_0, axis=1)
 
         h_0 = self.decoder_lat(z)
         # h_0 = h_0.unsqueeze(0).repeat(self.decoder_rnn.num_layers, 1, 1)
-        h_0 = lbann.Reshape(h_0, dims=str_list([1, 512]))
-        h_0 = lbann.Tessellate(h_0, dims=str_list((3, 512)))
+        h_0 = lbann.Reshape(h_0, dims=[1, 512])
+        h_0 = lbann.Tessellate(h_0, dims=(3, 512))
```
Comment (bvanessen) [inline applications/ATOM/models/wae.py:261]:
```suggestion
        h_0 = lbann.Tessellate(h_0, dims=[3, 512])
```

File: applications/graph/node2vec/model/random_projection.py [hunk 2]
```diff
@@ -10,46 +10,46 @@ def random_projection(indices, num_projections, projection_dim):
...
             ones,
-            scaling_factors=utils.str_list([2*(1-eps), -(1-eps)]),
+            scaling_factors=[2*(1-eps), -(1-eps)],
         )
     )
     proj = lbann.InstanceNorm(proj)
     proj = lbann.WeightedSum(
         proj,
-        scaling_factors=utils.str_list(1/projection_dim),
+        scaling_factors=1/projection_dim,
```
Comment (bvanessen) [inline applications/graph/node2vec/model/random_projection.py:52]:
```suggestion
        scaling_factors=[1/projection_dim],
```

**Comment by benson31** [comment]
CI looks good now (link in description is updated). Ready for review.

File: applications/graph/node2vec/model/random_projection.py [hunk 3]
```diff
@@ -10,46 +10,46 @@ def random_projection(indices, num_projections, projection_dim):
     # Note: proj_indices(i) = index*projection_dim + i
     proj_indices = lbann.WeightedSum(
         indices,
-        scaling_factors=utils.str_list(projection_dim),
+        scaling_factors=projection_dim,
     )
     iota = lbann.WeightsLayer(
-        dims=utils.str_list(projection_dim),
+        dims=.projection_dim,
```
Comment (bvanessen) [inline applications/graph/node2vec/model/random_projection.py:16]: @benson31 Is this period supposed to be at the start of the list?

Comment (benson31) [reply applications/graph/node2vec/model/random_projection.py:16]: Oh, probably not.

Comment (benson31) [inline applications/graph/node2vec/model/random_projection.py:16]:
```suggestion
        dims=projection_dim,
```

Comment (benson31) [reply applications/graph/node2vec/model/random_projection.py]: Fixed.

File: src/proto/factories/model_factory.cpp [hunk 4]
```diff
@@ -207,21 +207,24 @@ void assign_weights_to_objective_function(
     auto&& term = dynamic_cast<l2_weight_regularization*>(obj_terms[i]);
     if (term != nullptr) {
       ++num_l2_weight_regularization_terms;
-      const auto& params = proto_obj.l2_weight_regularization(num_l2_weight_regularization_terms-1);
+      const auto& params = proto_obj.l2_weight_regularization(
+        num_l2_weight_regularization_terms - 1);
       std::vector<ViewingWeightsPtr> term_weights;
-      for (auto&& weights_name : parse_list<std::string>(params.weights())) {
+      for (int wid = 0; wid < params.weights_size(); ++wid) {
```
Comment (bvanessen) [inline src/proto/factories/model_factory.cpp:213]: Why not keep this as an iterator over the params object?

File: src/proto/factories/model_factory.cpp [hunk 5]
```diff
@@ -155,11 +154,12 @@ void assign_weights_to_layers(
...
 
   // Find weights assigned to each layer
-  for (int i=0; i<proto_model.layer_size(); ++i) {
+  for (int i=0; i < proto_model.layer_size(); ++i) {
     const auto& proto_layer = proto_model.layer(i);
     auto layer_weights = layer_list[i]->get_weights_pointers();
     const bool is_frozen = layer_list[i]->is_frozen();
-    for (auto&& name : parse_list<std::string>(proto_layer.weights())) {
+    for (int ii = 0; ii < proto_layer.weights_size(); ++ii) {
+      auto const& name = proto_layer.weights(ii);
```
Comment (bvanessen) [inline src/proto/factories/model_factory.cpp:162]: Why not keep this as an iterator over the params object?

**Review by bvanessen — APPROVED** [review]
LGTM, I did have a few remaining questions.

**Pull request merged**